### DISCUSSION TITLE
[Kavita] Custom Library Browse-By/View-By Menu Options

### DISCUSSION THREAD

**Discussion by bnerickson** [Kareadita/Kavita#3445] [Ideas]
### Idea Description

Copied from https://github.com/Kareadita/Kavita/discussions/2556:

The current "Browse Authors" menu option is a great first step in allowing users to quickly jump into and find works to read by a parameter that is not its series.  However, by itself it displays a daunting amount of authors spread across every library.  I would like to be able to define sub-menus under each individual library menu for custom "Browse by X" metadata attributes.  Here's a mock-up to illustrate what I mean:

<img width="179" alt="browse_library_by_author" src="https://github.com/user-attachments/assets/ec1c5252-8983-4d08-afb7-8a100b8d903d">

In the example above, I would select the "Browse Authors" sub-menu under the "Books" library and a view identical to the "Browse Authors" global view would display; however, only the authors within the "Books" library would be displayed.

<img width="388" alt="browse_library_by_genre" src="https://github.com/user-attachments/assets/c0820c21-15f3-4e16-aa97-ea5300d09691">

In the example above, if I were to select the "Browse Genre" sub-menu under the "Manga" library, a page similar to the "Browse Authors" would display instead listing genres available in the "Manga" library.  Clicking on a "Genre" (say "Horror") would display only "Manga" works with the "Horror" genre.

Originally, I was only going to request "Browse Authors" sub-menus, but given there is at least one other feature request to browse by genres (https://github.com/Kareadita/Kavita/discussions/3173), I figured it might be better future-proofing to allow users to define their own sub-menus.

### Idea Category

User Experience

### Duration of Using Kavita

2 years

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by newyorker123** [comment]
I think the Browse page needs a Sort & Filter function, similar to the one on the All Series page. This would allow users to filter by Library, Role, and other criteria. Then, you can save the filter you create in nav using the existing Smart Filter feature. However, these would function parallel to other libraries rather than as true submenus.
